x-pack/elastic-agent/pkg/fleetapi/ack_cmd.go

```diff
@@ -79,24 +81,29 @@ func NewAckCmd(info agentInfo, client client.Sender) *AckCmd {
 
 // Execute ACK of actions to the Fleet.
 func (e *AckCmd) Execute(ctx context.Context, r *AckRequest) (*AckResponse, error) {
-	if err := r.Validate(); err != nil {
+	var err error
+	span, ctx := apm.StartSpan(ctx, "execute", "app.internal")
+	defer func() {
+		if err != nil {
+			apm.CaptureError(ctx, err).Send()
+		}
```

Comment:
[Suggestion]
It's safe to call `apm.CaptureError` with a nil error and then call `Send`. Therefore it can be simplified.
```suggestion
		apm.CaptureError(ctx, err).Send()
```